OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -443,33 +445,33 @@ protected void CancelProduction(string itemName, uint numberToCancel)
 
 		bool CancelProductionInner(string itemName)
 		{
-			var lastIndex = queue.FindLastIndex(a => a.Item == itemName);
+			var item = Queue.LastOrDefault(a => a.Item == itemName);
 
-			if (lastIndex > 0)
-				queue.RemoveAt(lastIndex);
-			else if (lastIndex == 0)
+			if (item != null)
 			{
-				var item = queue[0];
-
 				// Refund what has been paid
 				playerResources.GiveCash(item.TotalCost - item.RemainingCost);
-				FinishProduction();
+				EndProduction(item);
```

Comment:
Minor style nit (not your fault): You can remove the `else` branch by `return true;`ing here, and then unconditionally `return false;`ing outside the conditional.